app/grants/views.py

```diff
@@ -3960,16 +3960,14 @@ def clr_matches(request):
         if claim_tx is None:
             return Response({'message': 'claim_tx field is required!'}, status=400)
 
-        clr_matches = CLRMatch.objects.all()
-
-        clr_match = clr_matches.filter(pk=pk).first()
-
-        if not clr_match:
+        try:
+            clr_match = CLRMatch.objects.get(pk=pk)
+        except CLRMatch.DoesNotExist:
             return Response({'message': 'CLR Match not found!'}, status=404)
 
-        # update claim tx for all grant with grant.admin_address paid out by 
+        # update claim tx for all grant with grant.admin_address paid out by
         # the clr_match contract
-        clr_matches.filter(
+        CLRMatch.objects.filter(
```

Comment:
Is this more DB efficient ?
The older was written based off this 
https://docs.djangoproject.com/en/4.0/topics/db/queries/#retrieving-specific-objects-with-filters

Reply:
It will be the same performance, my intent here was to group more of the error scenarios which is looking for the specific clr_match, and then the query where we filter by the admin address and payout. I'm happy to revert this back if you preferred the previous code.

Reply:
@thelostone-mc @ksolo  i think it's more efficient... as we should avoid `objects.all()` queries